sdk/keyvault/test-resources.bicep

```diff
@@ -126,7 +126,7 @@ resource blobContainer 'Microsoft.Storage/storageAccounts/blobServices/container
   parent: blobService
 }
 
-resource managedIdentityRoleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+resource managedIdentityRoleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = if (enableHsm) {
```

Comment:
See here for the reference: https://github.com/Azure/azure-sdk-for-js/pull/32193/files#diff-acc1e9f7a7a2c752f6be82069d457afb346eadd2a07c0c4278bf99b67002266eR38-R41